packages/bot-skeleton/src/utils/strategy-helper.js

```diff
@@ -0,0 +1,72 @@
+/** Helper function to perform sequential execution of functions */
+export const pipe = (...fns) => {
+    return initialValue => {
+        return fns.reduce((nextParam, fn) => fn(nextParam), initialValue);
+    };
+};
+
+/** Helper function to parse block information from XML */
+export const extractBlocksFromXml = xml => {
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(xml, 'text/xml');
+    const blocks = xmlDoc.getElementsByTagName('block');
+
+    const blocksTypeAndFields = Array.from(blocks).map(block => {
+        const type = block.getAttribute('type');
+        const fields = Array.from(block.getElementsByTagName('field')).map(field => {
+            return {
+                name: field.getAttribute('name'),
+                value: field.textContent.trim(),
+            };
+        });
+        return { type, fields };
+    });
+
+    return blocksTypeAndFields;
+};
+
+/** Helper function to sort block childs based on type or field name */
+export const sortBlockChild = blocksArray =>
+    blocksArray.sort((a, b) => {
+        if (a.type < b.type) {
+            return -1;
+        }
+        if (a.type > b.type) {
+            return 1;
+        }
+        return JSON.stringify(a.fields) < JSON.stringify(b.fields) ? -1 : 1;
+    });
+
+/** Compare two blocks by field name and value */
+const areFieldsEqual = (fields1, fields2) => {
+    if (fields1?.length !== fields2?.length) {
+        return false;
+    }
+    for (let i = 0; i < fields1.length; i++) {
+        if (fields1[i]?.name !== fields2[i]?.name || fields1[i]?.value !== fields2[i]?.value) {
+            return false;
+        }
+    }
+    return true;
+};
+
+/** Sanitize XMLs and compare them */
+export const compareXml = (xml1, xml2) => {
+    // Extract block information from the XMLs and sort the blocks
+    const sortedBlocks1 = pipe(extractBlocksFromXml, sortBlockChild);
```

Comment:
here variable names are camelCase,  any reason? can you check , yeah our code base has a mix of casing. Just have a look at both files and keep whatever is suitable, but let know why.

Reply:
This one was a mistake. Thanks for catching Bhai!

Reply:
Bhai this one is actually a function. If you check the next lines you will see I am passing the XML into this function and storing it to the blocks1 and blocks2 variable. I am following the function to have camelCase and variable snake_case rule.